editoast/src/views/work_schedules.rs

```diff
@@ -510,4 +762,45 @@ pub mod tests {
 
         assert_eq!(work_schedule_project_response, expected);
     }
+
+    #[rstest]
+    async fn work_schedule_endpoints_workflow() {
+        let app = TestAppBuilder::default_app();
+
+        // Create a new group
+        let create_group_request = app.post("/work_schedules/group").json(&json!({}));
+        let group_creation_response = app
+            .fetch(create_group_request)
+            .assert_status(StatusCode::OK)
+            .json_into::<WorkScheduleGroupCreateResponse>();
+        let group_id = group_creation_response.work_schedule_group_id;
+        let work_schedule_url = format!("/work_schedules/group/{group_id}");
+
+        // Add a work schedule
+        let request = app.put(&work_schedule_url).json(&json!([{
+                "start_date_time": "2024-01-01T08:00:00Z",
+                "end_date_time": "2024-01-01T09:00:00Z",
+                "track_ranges": [],
+                "obj_id": "work_schedule_obj_id",
+                "work_schedule_type": "CATENARY"
+            }]
+        ));
+        app.fetch(request).assert_status(StatusCode::OK);
+
+        // Get the content of the group
+        let request = app.get(&work_schedule_url);
+        let work_schedules = app
+            .fetch(request)
+            .assert_status(StatusCode::OK)
+            .json_into::<Vec<WorkSchedule>>();
+        assert_eq!(1, work_schedules.len());
```

Comment:
Maybe check that the work schedule returned matches the work schedule sent in the put request and give one of its fields some random value (uuid in `obj_id` ?) to make sure it cannot be another pre-existing work schedule in the database ? I had issues in other tests where their behavior changed depending on the state of the local database when the tests assumed that a table was empty, that a value for a column with a unique constraint was not already taken etc.

Reply:
Fixed in 61426f6d6f5492a4585d822e904e34bccc38c9a9